flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -339,6 +343,13 @@ public void notifyCheckpointAborted(long checkpointId) throws Exception {
                             stateDesc.getClass(), this.getClass());
             throw new FlinkRuntimeException(message);
         }
+        RegisteredKeyValueStateBackendMetaInfo<N, S> meta =
```

Comment:
I think it should be `RegisteredKeyValueStateBackendMetaInfo<N, SV>`.

Moreover, if we think another type `SEV` instead of `SV` is necessary, I wonder can we change the code to 
~~~java
        RegisteredKeyValueStateBackendMetaInfo<N, SV, SEV> meta =
                new RegisteredKeyValueStateBackendMetaInfo<>(
                        stateDesc.getType(),
                        stateDesc.getName(),
                        namespaceSerializer,
                        stateDesc.getSerializer(),
                        snapshotTransformFactory);
~~~
and make class `RegisteredKeyValueStateBackendMetaInfo<N, S>` to `RegisteredKeyValueStateBackendMetaInfo<N, S, SEV>`?

Reply:
You're right, I'll change it to `SV`.

However, adding a third type argument (`SEV`) to Metadata requires changing a lot files while in most cases it will be either `?` or `SV`; and here it's not necessary as transformation is not used while logging. So I wouldn't add it.

Reply:
To be honest, the raw use of `RegisteredKeyValueStateBackendMetaInfo` constructor makes me a bit confused. 
I have tried to change `RegisteredKeyValueStateBackendMetaInfo<N, S>` to `RegisteredKeyValueStateBackendMetaInfo<N, S, SEV>` and it dose not need to change many files.  Or shall we change the code below?
~~~ java
        RegisteredKeyValueStateBackendMetaInfo<N, SV> meta =
                new RegisteredKeyValueStateBackendMetaInfo(
                        stateDesc.getType(),
                        stateDesc.getName(),
                        namespaceSerializer,
                        stateDesc.getSerializer(),
                        (StateSnapshotTransformer.StateSnapshotTransformFactory<SV>) snapshotTransformFactory);
~~~

Reply:
> Or shall we change the code below?
> (StateSnapshotTransformer.StateSnapshotTransformFactory<SV>) snapshotTransformFactory);

Good idea!